controller_manager/src/controller_manager.cpp

```diff
@@ -1603,10 +1603,25 @@ void ControllerManager::activate_controllers(
       {
         RCLCPP_ERROR(
           get_logger(),
-          "After activation, controller '%s' is in state '%s' (%d), expected '%s' (%d).",
+          "After activation, controller '%s' is in state '%s' (%d), expected '%s' (%d). Releasing "
+          "interfaces!",
           controller->get_node()->get_name(), new_state.label().c_str(), new_state.id(),
           hardware_interface::lifecycle_state_names::ACTIVE,
           lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
+        controller->release_interfaces();
+        // Now prepare and perform the stop interface switching as this is needed for exclusive
+        // interfaces
+        if (
+          !command_interface_names.empty() &&
+          (!resource_manager_->prepare_command_mode_switch({}, command_interface_names) ||
```

Comment:
We should not call this method here, as we are in the RT (update) loop, and this method is declared ad non-realtime relevant in the hardware description, as it is usually called from the non-rt loop (service callback).

Reply:
I agree with you! I completely forgot that this is a non-RT method. We can do it at the end as you mentioned in the other comment